querybook/server/lib/query_analysis/templating.py

```diff
@@ -34,15 +33,58 @@ class LatestPartitionException(QueryTemplatingError):
     pass
 
 
-# The first part is regex for single line comment, ie -- some comment
-# the second part is for multi line comment, ie /* test */
-comment_re = re.compile(r"((?:--.*)|(?:\/\*(?:.|\n)*?\*\/))", re.MULTILINE)
-
-
 def _escape_sql_comments(query: str):
-    return re.sub(
-        comment_re, lambda match: "{{ " + json.dumps(match.group()) + " }}", query
-    )
+    result = []
+    pos = 0
+    n = len(query)
+    while pos < n:
+        ch = query[pos]
+        # If inside a quoted string, just copy it verbatim.
+        if ch in ("'", '"'):
+            quote_char = ch
+            start = pos
+            pos += 1
+            while pos < n:
+                if query[pos] == "\\":
+                    pos += 2
```

Comment:
add a comment here explaining we are skipping 2 chars because its \' or \"

Reply:
actually ANSI only uses single quote and uses doubled single quote to escape the single quote. will update